src/clientlayers/RetryRequest.jl

```diff
@@ -92,9 +93,11 @@ end
 
 function no_retry_reason(ex, req)
     buf = IOBuffer()
+    unwrapped_ex = unwrap_exception(ex)
```

Comment:
I don't think we need this `unwrap_exception` if we're doing it via dispatch in `isrecoverable`; so I think we remove it here and keep it above.

Reply:
Oh, but this is just for the `no_retry_reson`, so maybe we just need this for printing, which is fine.

Reply:
Exactly yeah, this is for printing. Thanks!